lib/Doctrine/MongoDB/Aggregation/Expr.php

```diff
@@ -0,0 +1,890 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\MongoDB\Aggregation;
+
+use LogicException;
+
+/**
+ * Fluent interface for adding operators to aggregation stages.
+ *
+ * @author alcaeus <alcaeus@alcaeus.org>
+ */
+class Expr
+{
+    /**
+     * @var array
+     */
+    protected $expr = array();
+
+    /**
+     * The current field we are operating on.
+     *
+     * @var string
+     */
+    protected $currentField;
+
+    /**
+     * Adds numbers together or adds numbers and a date. If one of the arguments is a date, $add treats the other arguments as milliseconds to add to the date.
+     *
+     * The arguments can be any valid expression as long as they resolve to either all numbers or to numbers and a date.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/add/
+     * @param mixed|self $expression1
+     * @param mixed|self $expression2
+     * @param mixed|self $expression3,... Additional expressions
+     * @return $this
```

Comment:
I changed it to ```@return self``` now. For the record, in many IDEs this makes a difference in auto-completion. Using ```$this``` instead of ```self``` provides correct auto-completion when extending other classes.